src/TreeViewList/TreeViewList.tsx

```diff
@@ -115,6 +165,30 @@ const TreeViewList = ({
   );
 };
 
+/**
+ * Recursively finds a node by its ID.
+ *
+ * @param nodes - The nodes to search.
+ * @param nodeId - The ID of the node to find.
+ * @returns The node with the specified ID, or null if not found.
+ */
+const getNodeById = (
+  nodes: TreeNodeItem[],
+  nodeId: string
+): TreeNodeItem | null => {
+  for (const node of nodes) {
+    if (node.nodeId === nodeId) {
+      return node;
+    } else if (node.children) {
+      const result = getNodeById(node.children, nodeId);
+      if (result) {
+        return result;
+      }
+    }
+  }
+  return null;
+};
+
```

Comment:
Is it the most performant way to be calling this recursive function every time a selection is made? We already map through all of the nodes on initial render to build the tree so why not just store a map of child node ids once then? I imagine indexing into a map is better than calling this function each time.

Reply:
I suppose it's also questionable whether this functionality should live in the component itself or whether the consuming service just deals with it. If the consuming service only cares about node selection for certain nodes (child nodes being one such example) then just store a map yourself when you build the tree data structure and use that in your own onNodeSelect handler.

Reply:
We profiled this together and with the largest dataset we have (DATA) it seems to only be 2-5ms with this implementation so we keep it for now.